src/codeu/chat/server/Controller.java

```diff
@@ -103,6 +100,107 @@ public Message newMessage(Uuid id, Uuid author, Uuid conversation, String body,
   }
 
   @Override
+  public boolean deleteMessage(Uuid msg, Uuid conversation) {
```

Comment:
Not exactly. This method calls a method delete(Message message) written in the Model class, which deletes the given Message from the three Stores on Model that keep track of existing Messages. These Stores on the server are eventually indirectly used by ClientMessage to update the client side list of existing messages. The iterators are there to help parse through the messageByTime Store, so I can update the firstMessage and lastMessage fields in Conversation appropriately if the Message to be deleted is the first message in the conversation, the last message in the conversation, or both. I've added comments to the helper methods explaining their purpose.